packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js

```diff
@@ -55,6 +59,19 @@ export default function SidebarNavigationScreenNavigationMenus() {
 
 	const hasNavigationMenus = !! navigationMenus?.length;
 
+	// If there is a **single** menu, update the state
+	// to indicate the editing of that menu. This
+	// will trigger focus mode for the given menu.
+	useEffect( () => {
+		if ( navigationMenus?.length === 1 ) {
+			setNavigationMenu( navigationMenus[ 0 ].id );
+		}
+
+		return () => {
+			setNavigationMenu( null );
```

Comment:
There is a bug here which means that when you go back from Navigation focus mode the editor doesn't know which post type it's editing.

If you remove it it still doesn't work because the logic in `packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js` doesn't seem to account for this.